### PR TITLE
Circles Rebuild - Part XI - US Contributions Page

### PR DESCRIPTION
**Pull request by Ap0c**
## Why are you doing this?

Creating a circles version of the US contributions landing page.

[**Trello Card**](https://trello.com/c/aLdmucS4/1286-contribute-page-us)

cc @JustinPinner 

## Changes

- Created an Application method for the US contributions page, to allow viewing the circles designs via query param.
- Created the new page, with an introduction and contribute section.
- Marked the old US contributions page as 'old'.
- Tweaked the Contribution Selection reducer to allow overriding initial state at the page level.

## Screenshots

**Desktop - Monthly:**

us-desktop-monthly

**Desktop - One-off:**

us-desktop-oneoff

**Mobile:**

us-mobile

### CODE REVIEW

File: app/controllers/Application.scala [hunk 1]
```diff
@@ -101,6 +101,26 @@ class Application(
...
+    } else {
+      Ok(views.html.contributionsLanding(
+        title,
+        description = Some(stringsConfig.contributionLandingDescription),
+        id,
+        js,
+        contributionsPayPalEndpoint
+      ))
+    }
+  }
```
Comment (hosamaly) [inline app/controllers/Application.scala]: Do you think it's a good idea to reduce the duplication? I mean, both branches are the same except for `id` and `js`...

File: app/controllers/Application.scala [hunk 2]
```diff
@@ -101,6 +101,26 @@ class Application(
     }
   }
 
+  def contributionsLandingUS(title: String, id: String, js: String, newDesigns: String): Action[AnyContent] = CachedAction() { implicit request =>
+    if (newDesigns == "circles-garnett") {
```
Comment (hosamaly) [inline app/controllers/Application.scala]: I've seen multiple PRs with this condition. Is the value `circles-garnett` defined as a constant somewhere?

File: app/controllers/Application.scala (hunk 1 above)
Comment (Ap0c) [reply app/controllers/Application.scala]: What would be a good way to do this in Scala, ternary expression?

File: app/controllers/Application.scala (hunk 2 above)
Comment (Ap0c) [reply app/controllers/Application.scala]: Hmm, could define a `val` for this scoped to the `Application` class?

File: app/controllers/Application.scala (hunk 1 above)
Comment (hosamaly) [reply app/controllers/Application.scala]: Scala doesn't have ternary expressions, but perhaps something like this?
```scala
def contributionsLandingUS(title: String, id: String, js: String, newDesigns: String): Action[AnyContent] = CachedAction() { implicit request =>
  def contributionsLanding(pageId: String, pageJs: String) = Ok(views.html.contributionsLanding(
    title,
    description = Some(stringsConfig.contributionLandingDescription),
    pageId,
    pageJs,
    contributionsPayPalEndpoint
  )

  if (newDesigns == "circles-garnett")
    contributionsLanding(pageId = "contributions-landing-page-us", pageJs = "contributionsLandingPageUS.js")
  else
    contributionsLanding(id, js)
}

File: app/controllers/Application.scala (hunk 2 above)
Comment (hosamaly) [reply app/controllers/Application.scala]: Sounds good.

File: app/controllers/Application.scala (hunk 1 above)
Comment (hosamaly) [reply app/controllers/Application.scala]: I've just noticed that this is practically a duplicate of `contributionsLandingUK`. It looks like there is even more room to deduplicate them.

Comment (Ap0c) [reply app/controllers/Application.scala]: Yeah it's a test across multiple pages. Ok I've had a chat with a couple people offline, I'm going to do a follow-up PR that changes all instances of this to something a bit cleaner. Thanks for the feedback! 👍

**Pull request merged**